plugins/needs_restarting.py

```diff
@@ -277,15 +279,28 @@ def run(self):
         NEED_REBOOT.extend(opt)
         if self.opts.reboothint:
             need_reboot = set()
+            need_reboot_depends_on_dbus = set()
             installed = self.base.sack.query().installed()
             for pkg in installed.filter(name=NEED_REBOOT):
                 if pkg.installtime > process_start.boot_time:
                     need_reboot.add(pkg.name)
-            if need_reboot:
+            base = dnf.Base()
```

Comment:
Do we need a new `base` to get the installed packages?
 I see on line 283: `installed = self.base.sack.query().installed()`